src/shared/proto/robot_component_status.proto

```diff
@@ -0,0 +1,176 @@
+    WHEEL_1_ENCODER_NOT_COMMUTATING = 12;
+    WHEEL_2_ENCODER_NOT_COMMUTATING = 13;
+    WHEEL_3_ENCODER_NOT_COMMUTATING = 14;
+    WHEEL_0_HALL_SENSOR_STUCK_LOW   = 15;
+    WHEEL_1_HALL_SENSOR_STUCK_LOW   = 16;
+    WHEEL_2_HALL_SENSOR_STUCK_LOW   = 17;
+    WHEEL_3_HALL_SENSOR_STUCK_LOW   = 18;
+    DRIBBLER_HALL_SENSOR_STUCK_LOW  = 19;
+    WHEEL_0_HALL_SENSOR_STUCK_HIGH  = 20;
+    WHEEL_1_HALL_SENSOR_STUCK_HIGH  = 21;
+    WHEEL_2_HALL_SENSOR_STUCK_HIGH  = 22;
+    WHEEL_3_HALL_SENSOR_STUCK_HIGH  = 23;
+    DRIBBLER_HALL_SENSOR_STUCK_HIGH = 24;
+    // Edge triggered events
+    ICB_CRC_ERROR                        = 25;
+    RECEIVE_FRAME_CHECK_SEQUENCE_FAILURE = 26;
+    CRASHED_CORE_DUMPED                  = 27;
+    CRASHED_NO_CORE_DUMP                 = 28;
+
+    /** 2014 DONGLE Messages **/
+    ESTOP_BROKEN_MESSAGE        = 29;
+    RX_FCS_FAIL_MESSAGE         = 30;
+    SECOND_DONGLE_MESSAGE       = 31;
+    TRANSMIT_QUEUE_FULL_MESSAGE = 32;
+    RECEIVE_QUEUE_FULL_MESSAGE  = 33;
+}
+
+/* Firmware build information and primitive status  */
+message FirmwareStatus
+{
+    // The microcontroller firmware build ID.
+    required uint32 fw_build_id = 1;
+
+    // Whether the robot is in direct mode.
+    required bool direct_control = 2;
+
+    // Whether or not the build ID information is valid.
+    required bool build_ids_valid = 3;
+
+    /** 2014 robots-specific fields**/
+    // The FPGA bitstream build ID.
+    optional uint32 fpga_build_id = 4;
+}
+
+/* Data about the status of the motors */
+message MotorStatus
+{
+    // The speed of the robot’s dribbler motor, in revolutions per minute.
+    required int32 dribbler_speed = 1;
+
+    // The maximum power level understood by the direct_dribbler function.
+    required uint32 direct_dribbler_max = 2;
+
+    /* Frankie Robot Fields */
+    // Encoder information
+    optional EncoderData encoder_data = 3;
+}
+
+/* Data about the network connection with the robots, including network-derived values */
+message NetworkStatus
+{
+    // When the proto was created according to lwIP SNTP time
+    required uint64 time_created = 1;
+
+    /* Frankie Robot Fields */
+    // Indicates the time elapsed since the last vision packet was received in ms
+    // UINT64_MAX indicates no vision packet was ever received
+    optional uint64 time_since_last_vision_received = 2;
+
+    // Indicates the time elapsed since the last primitive packet was received in ms
+    // UINT64_MAX indicates no primitive packet was ever received
+    optional uint64 time_since_last_primitive_received = 3;
+
+    /** 2014 robots-specific fields**/
+    // The link quality of the last received packet, from 0 (worst) to 1 (best).
+    optional double link_quality = 4;
+
+    // The received signal strength of the last received packet, in decibels.
+    optional int32 received_signal_strength_db = 5;
+
+    // Whether or not the robot is currently responding to radio communication.
+    optional bool alive = 6;
```

Comment:
I don't think this field is useful. It should be pretty obvious if the robot is alive if the AI is receiving status messages at all, I don't think an extra bool is useful.